### DISCUSSION TITLE
Displaying "empty" slides when the number of slides is not divisible by slidesToScroll #583

### DISCUSSION THREAD

**Discussion by lockr7** [davidjerleke/embla-carousel#584] [Help] [answered]
### Feature request is related to

- [x] embla-carousel (core package)
- [x] embla-carousel-react

### Is your feature request related to an issue?

- Currently when I have slidesToScroll = 3 but only 5 slides (or any other combination, like slidesToScroll = 5 but only 8 slides), I get this behavior where it doesn't actually scroll 3 slides but only scrolls until the 5th slide is the last displayed

<img width="946" alt="image" src="https://github.com/davidjerleke/embla-carousel/assets/22669712/5a6969f4-e701-4f72-9085-b49c7d3af96d">

### Describe the solution you'd like

- When I have slidesToScroll = 3 but only 5 slides (or any other combination, like slidesToScroll = 5 but only 8 slides), I want the first 3 slides to be visible and then on drag or on next button click, I want only the 4th and 5th slide to be visible with the 6th slot left empty/blank

<img width="868" alt="image" src="https://github.com/davidjerleke/embla-carousel/assets/22669712/0bebea4a-6056-4c71-b9c0-655d2de8c18a">

### Describe alternatives you've considered

- This may already be possible, but I couldn't figure it out. I tried changing containScroll but didn't seem to help.

**Comment by davidjerleke** [comment, marked as answer]
Hi @lockr7,

If I understand you correctly, you want to align the slides to the left edge of the viewport. You can try setting your options to:

```ts
const options = {
  containScroll: false, // allow leading and trailing space
  align: 'start', // align to start edge
  slidesToScroll: 3
}
``` 

Please note that you can set [`slidesToScroll`](https://www.embla-carousel.com/api/options/#slidestoscroll) to `'auto'` and Embla will group your slides automatically based on how many slides that fit into the viewport. It may be of interest to you.

Best,
David

**Reply by davidjerleke** [reply]
@lockr7 have you seen my response?

**Reply by lockr7** [reply]
yes! it worked, thanks very much
